### PR TITLE
Fix divisions when joining two single-partition dataframes

### PR DESCRIPTION
**Pull request by bluecoconut**
There was an error where if `npartition=1` for both `left` and `right` on a merge, then the check inside of `single_partition_join` would short to using the `right` divisions, even if it was supposed to be `left` table's divisions that are used.

- [x] Tests added / passed
- [x] Passes `flake8 dask`

Note: the test I added first validated the bug (on previous code), and then was resolved with the fix in `single_partition_join`

This answers the bug identified in https://github.com/dask/dask/issues/4617

### CODE REVIEW

File: dask/dataframe/multi.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def single_partition_join(left, right, **kwargs):
         else:
             divisions = [None for _ in right.divisions]
 
-    elif right.npartitions == 1:
+    elif right.npartitions == 1 and kwargs['how'] in ('inner', 'left'):
```
Comment (mrocklin) [inline dask/dataframe/multi.py]: Are there cases where neither of these two branches is taken?

**Comment by mrocklin** [comment]
Thanks for taking this on @bluecoconut !  Sorry for the slow response on your issue.  A combination of conferences and newborns I think is saturating most maintainer cycles these days :)

File: dask/dataframe/multi.py (hunk 1 above)
Comment (bluecoconut) [reply dask/dataframe/multi.py]: These are the conditions required to enter `single_partition_join` method. 

`dask.dataframe.multi.py#370-371`

Comment (bluecoconut) [reply dask/dataframe/multi.py]:
```    elif (left.npartitions == 1 and how in ('inner', 'right') or
          right.npartitions == 1 and how in ('inner', 'left')):	          right.npartitions == 1 and how in ('inner', 'left')):
        return single_partition_join(left, right, how=how, right_on=right_on,	        return single_partition_join(left, right, how=how, right_on=right_on,
                                     left_on=left_on, left_index=left_index,	                                     left_on=left_on, left_index=left_index,
                                     right_index=right_index,	                                     right_index=right_index,
                                     suffixes=suffixes, indicator=indicator)	                                     suffixes=suffixes, indicator=indicator)
```

Comment (bluecoconut) [reply dask/dataframe/multi.py]: Could add (if desired for future developer ease) an `else: raise NotImplementedError...` Not sure what best practices in this code base are. At the moment, the method is only called inside of that parent `elif` statement.

Comment (mrocklin) [reply dask/dataframe/multi.py]: Ah, ok, sounds like it's safe then.  An `else: raise` block seems reasonable to me.  I'm happy to defer to your preferences here.

Comment (bluecoconut) [reply dask/dataframe/multi.py]: Added the raise error~

**Comment by mrocklin** [comment]
Thanks @bluecoconut !

**Pull request merged**